perceval/utils/dist_metrics.py

```diff
@@ -50,3 +50,32 @@ def tvd_dist(dist_lh: BSDistribution, dist_rh: BSDistribution) -> float:
     tvd = 0.5 * sum(abs(dist_lh.get(basic_state, 0) - dist_rh.get(basic_state, 0)) for basic_state in all_states)
 
     return tvd
+
+
+def kl_divergence(ideal_dist: BSDistribution, est_dist: BSDistribution) -> float:
+    """
+    Computes the Kullback-Leibler (KL) divergence of a model (simulated/observed) BSdistribution with respect to
+     an ideal BSDistribution.
+     Our computation ignores states absent from the estimated distribution or have null probabilities.
+
+    :param ideal_dist: Ideal BSDistribution (known from theory or an ideal computation)
+    :param est_dist: Estimated BSDistribution (simulated or observed from experiment)
+    :return : KL divergence of the estimated distribution relative to the ideal.
+    """
+
+    kl_div = 0
+    zero_states_count = 0  # states with null probabilities or missing in estimated distribution
+
+    for states, ideal_probs in ideal_dist.items():
```

Comment:
Variables you retrieve by iterating on the distribution should be singular : `state, ideal_prob`

Below `est_prob`